tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -258,82 +238,96 @@ spec:
       mode: DISABLE`,
 					expected: []expect{
 						{
-							port:   8085,
-							schema: protocol.HTTP,
-							want:   false,
+							port: &echo.Port{ServicePort: 8085, Protocol: protocol.HTTP},
+							want: false,
 						},
 						{
-							port:   8086,
-							schema: protocol.HTTP,
-							want:   true,
+							port: &echo.Port{ServicePort: 8086, Protocol: protocol.HTTP},
+							want: true,
 						},
 						{
-							port:   8087,
-							schema: protocol.TCP,
-							want:   false,
+							port: &echo.Port{ServicePort: 8087, Protocol: protocol.TCP},
+							want: false,
 						},
 						{
-							port:   8088,
-							schema: protocol.TCP,
-							want:   true,
+							port: &echo.Port{ServicePort: 8088, Protocol: protocol.TCP},
+							want: true,
 						},
 					},
 				},
 			}
 
-			for _, cluster := range ctx.Clusters() {
-				client := apps.Naked.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
-				destination := apps.A.Match(echo.Namespace(ns.Name())).GetOrFail(ctx, echo.InCluster(cluster))
-				from := getWorkload(client, ctx)
-				destWorkload := getWorkload(destination, ctx).Address()
-				for _, tc := range cases {
-					ctx.NewSubTest(fmt.Sprintf("In %s/%v", cluster.StableName(), tc.name)).Run(func(ctx framework.TestContext) {
-						ctx.Config().ApplyYAMLOrFail(ctx, ns.Name(), tc.config)
-						util.WaitForConfig(ctx, tc.config, ns)
+			// srcFilter finds the naked app as client.
+			findNaked := func(instances echo.Instances) echo.Instances {
+				return apps.Naked.Match(echo.Service(util.NakedSvc))
+			}
+			srcFilter := []echotest.SimpleFilter{findNaked}
+			// dstFilter finds the app a and vm as destination.
+			findPodAndVM := func(instances echo.Instances) echo.Instances {
```

Comment:
Two main concerns with the filtering:

1. Filters should be _exclusive_. We want to gain coverage as new types of workload are added, without modifying the tests. To do this, we only want to remove things that we know won't work, rather than only include the things that already work. 

If the full set of apps is: `a, b, c, naked, headless, external, vm`, and you only need one pod, and it doesn't make sense to test headless/external/naked as destinations, then remove each of those. 

2. Filters should look at the properties of the given instances, rather than the names. 

```go
// do this
func (instances echo.Instances) {
   return instances.Match(echo.IsVirtualMachine())
}

// avoid this
func (_ echo.Instances) {
   return apps.VM
}
```

What you want to test is:
- only one pod (for example, a, b, c being pods, just use a)
- don't have a headless or naked service as the destination


We can do this as:

```
.To(
  SingleSimplePodServiceAndAllSpecial(),
  echo.Not(func(instances echo.Instances) { return instances.Match(echo.IsHeadless()) }),
  echo.Not(func(instances echo.Instances) { return instances.Match(echo.IsNaked()) }),
  echo.Not(func(instances echo.Instances) { return instances.Match(echo.IsExternal()) }),
)
```

Reply:
I have updated the apps with exclusive filters that filter workloads by properties rather than names.